src/odemis/driver/xt_client.py

```diff
@@ -1607,6 +1608,7 @@ def _updatePixelSize(self):
 
     def _setDwellTime(self, dwell_time):
         self.parent.set_dwell_time(dwell_time)
+        self._current_dwell_time = dwell_time
         return self.parent.get_dwell_time()
```

Comment:
It's still going to indicate that the dwell time hasn't changed, as you returned the "accepted" value. Or does the server return an acceptable value already?  If not, that's confusing as the user could set the dwell time to a new value, the old value would be returned, and later when changing to external, suddenly the requested value is set.
I think it should just return the value that is "about to be set as soon as external will be disabled". (ie, check if external is True, and if so, don't ask the server the new value.

Reply:
> I think it should just return the value that is "about to be set as soon as external will be disabled". (ie, check if external is True, and if so, don't ask the server the new value.

Thanks, this makes sense

Reply:
Just for safety, could you still call `self.parent.set_dwell_time(dwell_time)`. This way, if for some systems, it has effect, it will still behave as before.